submissions/api.py

```diff
@@ -447,27 +447,27 @@ def get_all_course_submission_information(course_id, item_type, read_replica=Tru
     if read_replica:
         submission_qs = _use_read_replica(submission_qs)
 
-    query = submission_qs.select_related('student_item').prefetch_related('score_set').filter(
+    query = submission_qs.select_related('student_item__scoresummary__latest__submission').filter(
```

Comment:
If we stick with latest score, the comment for this method should mention "latest score" each time it currently reads "score".  The function comment should also mention that an empty score will be sent for submissions with no score.